### DISCUSSION TITLE
Is it an API like model.summary in Pytorch?

### DISCUSSION THREAD

**Discussion by BraveDrXuTF** [google/flax#3599] [Q&A] [answered]
My need is to know the exact parameter amount of a model, I can use a function like this  

```
def count_array_shapes(dictionary):  
    shapes = []  
    for value in dictionary.values():  
        if isinstance(value, dict):  
            shapes.extend(count_array_shapes(value))  
        elif isinstance(value, np.ndarray):  
            shapes.append(tuple(value.shape))  
    return shapes 
```
to get the shape of parameters, and then I can calculate them to get the parameter amount. So I wonder if there is an API to do this, or I can contribute some code as an API?

**Comment by cgarciae** [comment, marked as answer]
Hey! Take a look at [Module.tabulate](https://flax.readthedocs.io/en/latest/api_reference/flax.linen/module.html#flax.linen.Module.tabulate).

**Reply by BraveDrXuTF** [reply]
Got it, thanks.
